torch/csrc/autograd/python_function.cpp

```diff
@@ -78,11 +115,18 @@ namespace autograd {
 // NOTE: this function is written in a way that assumes it's only called for
 // backward; it's used by engine.cpp.  This is responsible for forwarding a call
 // from C++'s Node::apply to a Python method "apply".
-auto PyNode::apply(variable_list&& inputs) -> variable_list {
+auto PyNode::apply(variable_list&& inputs, std::optional<PyObject*> compiler)
```

Comment:
Rather than changing `apply()` you should use `apply_with_saved()`

Reply:
PyNode::apply_with_saved calls into PyNode::apply (which calls the backward). There's a lot of shared code for this case, should we fork the implementations with helpers?